packages/ethereum-contracts/test/scripts/deployment.test.js

```diff
@@ -210,14 +211,19 @@ contract("Embeded deployment scripts", (accounts) => {
                     nonUpgradable: true,
                     useMocks: false,
                 });
-                await expectRevertedWith(
-                    deployFramework(errorHandler, {
+                try {
+                    await deployFramework(errorHandler, {
                         ...deploymentOptions,
                         nonUpgradable: true,
                         useMocks: true, // force an update attempt
-                    }),
-                    "SF: non upgradable"
-                );
+                    });
+                } catch (err) {
+                    if (process.env.IS_TRUFFLE) {
```

Comment:
follow up: we cannot use the expectCustomError logic because the return value/error of the deployFramework code doesn't interface with the expectCustomError code by hardhat.